### PR TITLE
Add darwin skeletons

### PR DESCRIPTION
**Pull request by egernst**
Add initial stubs for darwin specific handling for network and resource control.

### CODE REVIEW

File: src/runtime/virtcontainers/network_darwin.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+type DarwinNetwork struct {
+	networkID         string
+	interworkingModel NetInterworkingModel
+	networkCreated    bool
+	eps               []Endpoint
+}
+
+func NewNetwork(configs ...*NetworkConfig) (Network, error) {
+	if len(configs) > 1 {
+		return nil, fmt.Errorf("Too many network configurations")
```
Comment (dcantah) [inline src/runtime/virtcontainers/network_darwin.go:30]: Prefer `errors.New` if no formatting (also go idioms is to start errors lowercased)

File: src/runtime/virtcontainers/network_darwin.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+	}
+
+	// Empty constructor
+	if len(configs) == 0 {
+		return &DarwinNetwork{}, nil
+	}
+
+	config := configs[0]
+	if config == nil {
+		return nil, fmt.Errorf("Missing network configuration")
```
Comment (dcantah) [inline src/runtime/virtcontainers/network_darwin.go:40]: Prefer `errors.New` if no formatting (also go idioms is to start errors lowercased)

File: src/runtime/pkg/resourcecontrol/cgroups_darwin.go [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+import (
+	"fmt"
+
+	"github.com/opencontainers/runtime-spec/specs-go"
+)
+
+type DarwinResourceController struct{}
+
+func RenameCgroupPath(path string) (string, error) {
+	return "", fmt.Errorf("RenameCgroupPath not supported on Darwin")
```
Comment (dcantah) [inline src/runtime/pkg/resourcecontrol/cgroups_darwin.go:17]: Prefer `errors.New` if no formatting (also go idiom is to start errors lowercased)

**Review by dcantah — COMMENTED** [review]
LGTM after some nits on errors

File: src/runtime/virtcontainers/network_darwin.go (hunk 1 above)
Comment (egernst) [reply src/runtime/virtcontainers/network_darwin.go]: ack -- I'll keep non-idiomatic just based on it being the function name?

**Comment by egernst** [comment]
Thanks Danny -- good call.  Updated.

File: src/runtime/pkg/resourcecontrol/cgroups_darwin.go (hunk 3 above)
Comment (egernst) [reply src/runtime/pkg/resourcecontrol/cgroups_darwin.go]: ack -- I'll keep non-idiomatic just based on it being the function name?

File: src/runtime/virtcontainers/network_darwin.go (hunk 1 above)
Comment (dcantah) [reply src/runtime/virtcontainers/network_darwin.go]: Oh I mean the error string haha so "too many network configurations"*

Comment (dcantah) [reply src/runtime/virtcontainers/network_darwin.go]: https://github.com/golang/go/wiki/Errors#errors

**Review by fengwang666 — APPROVED** [review]
LGTM

**Review by fidencio — APPROVED** [review]
lgtm, thanks @egernst!

**Pull request merged**